data/handle_data.go

```diff
@@ -177,11 +185,18 @@ func WriteDataToFile(filename string, testData map[string][]libunlynx.DpClearRes
 	}
 
 	writer := bufio.NewWriter(fileHandle)
+	if err := writer.Flush(); err != nil {
+		log.Fatal(err)
```

Comment:
like above: it's much cleaner to return an error and let the caller decide what should happen...